ignition/service/logging.py

```diff
@@ -58,10 +70,42 @@ def format(self, record):
         result = self._obfuscate_sensitive_data(result)
         return result
 
+    def _mask_sensitive_data(self, suffix, prefix, message, rep=None):
+        regex = re.compile('{0}(.*?){1}'.format(prefix, suffix), flags=re.DOTALL | re.IGNORECASE)
+        abc = re.findall(regex, message)
+        if len(abc) > 0:
+            if rep is None:
+                rep = '*' * len(abc[0])
+                if abc[0][0] != "'" and abc[0][len(abc[0])-1] == "'":
+                    rep = ":" + rep + "'"
+                elif abc[0].find("'") != -1:
+                    rep = "':'" + rep + "'"
+                elif abc[0].find('"') != -1:
+                    rep = '":"' + rep + '"'
+                else:
+                    rep = ":" + rep
+            masked_value = prefix + rep + suffix
+            message = re.sub(regex, masked_value, message)
+        return message
+
     def _obfuscate_sensitive_data(self, record_message):
         if record_message is None:
             return record_message
-        return re.sub(PRIVATE_KEY_REGEX, OBFUSCATED_PRIVATE_KEY, record_message)
+        
+        pwd_message = self._mask_sensitive_data(PWD_SUFFIX_WITH_NEXTLINE, PWD_PREFIX, record_message)
+        if pwd_message == record_message:
+            pwd_message = self._mask_sensitive_data(PWD_SUFFIX, PWD_PREFIX, record_message)
+
+        username_message = self._mask_sensitive_data(PWD_SUFFIX_WITH_NEXTLINE, USERNAME_PREFIX, pwd_message)
+        if username_message == pwd_message:
+            username_message = self._mask_sensitive_data(PWD_SUFFIX, USERNAME_PREFIX, pwd_message)
+
+        replacements=[(PRIVATE_KEY_PREFIX, PRIVATE_KEY_SUFFIX, OBFUSCATED_PRIVATE_KEY), (TOKEN_PREFIX, TOKEN_SUFFIX, None),
+                      (BEGIN_CERT_PREFIX, END_CERT_SUFFIX, OBFUSCATED_CERTIFICATE)]
+        for pre,suf,rep in replacements:
+            username_message = self._mask_sensitive_data(suf, pre, username_message, rep)
```

Comment:
Use generic name instead of username_message here as this has message with all the confidential data masked.

Reply:
Modified